consensus/core/src/commit_finalizer.rs

```diff
@@ -683,6 +697,33 @@ impl CommitFinalizer {
         finalized_transactions
     }
 
+    // Returns true if the pending block round can be GC'ed when proposing blocks in current commit.
+    fn check_gc_at_current_commit(
+        gc_rounds: &[(CommitIndex, Round)],
+        pending_block_round: Round,
+        pending_commit_index: CommitIndex,
+        current_commit_index: CommitIndex,
+    ) -> bool {
+        assert!(
+            pending_commit_index <= current_commit_index,
+            "Pending {pending_commit_index} should be <= current {current_commit_index}"
+        );
+        if pending_commit_index == current_commit_index {
+            return false;
+        }
+        // Since GC round only advances after a commit, when proposing blocks for the current commit,
+        // the GC round is determined by the commit previous to the current commit.
```

Comment:
Similar I think we should give a bit more context here on the scenario of "proposing".